### PR TITLE
Remove the DAHU test

### PR DESCRIPTION
**Pull request by kjetilk**
Following https://github.com/solid/conformance-test-harness/issues/155 I'm proposing the remove the corner case evil `DAHU` method test to. Instead, this has a `TRACE` test, which is in HTTP but not in Solid. 

Unfortunately, it is hard to have a conditional test now, and previously, I also checked the `Allow` header as that should be on every `405` but with this variability, it is hard to do, so I removed it for now.

### CODE REVIEW

**Review by RubenVerborgh — APPROVED** [review]
Thanks—fine with me.
Just not sure what _Solid_ requirement we are testing here, but that's part of another discussion.

File: protocol/read-write-resource/method-not-allowed.feature [hunk 1]
```diff
@@ -1,18 +1,19 @@
-Feature: Respond with 405 for non-existent method
+Feature: Error for undefined method
```
Comment (edwardsph) [inline protocol/read-write-resource/method-not-allowed.feature:1]:
```suggestion
Feature: Error for unsupported method
```
Since TRACE is valid for some servers, undefined seems wrong.

File: protocol/read-write-resource/method-not-allowed.feature [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
     * def testContainer = rootTestContainer.createContainer()
     * def resource = testContainer.createResource('.ttl', karate.readAsString('../fixtures/example.ttl'), 'text/turtle');
 
-  Scenario: Check response for DAHU method on resource
-    * def response = clients.alice.sendAuthorized('DAHU', resource.url, null, null, 'HTTP_1_1')
-    Then assert response.status == 405
-    And assert response.headers.allow != null
+  Scenario: Check response for TRACE method on resource
+    * def response = clients.alice.sendAuthorized('TRACE', resource.url, null, null, null)
+    Then assert response.status == 400 || response.status == 405 || response.status == 501
```
Comment (edwardsph) [inline protocol/read-write-resource/method-not-allowed.feature:9]:
```suggestion
    Then match [400, 405, 501] contains response.status
```

File: protocol/read-write-resource/method-not-allowed.feature [hunk 3]
```diff
@@ -1,18 +1,19 @@
...
+#    And assert response.headers.allow != null
 
-  Scenario: Check response for DAHU method on container
-    * def response = clients.alice.sendAuthorized('DAHU', testContainer.url, null, null, 'HTTP_1_1')
-    Then assert response.status == 405
-    And assert response.headers.allow != null
+  Scenario: Check response for TRACE method on container
+    * def response = clients.alice.sendAuthorized('TRACE', testContainer.url, null, null, null)
+    Then assert response.status == 400 || response.status == 405 || response.status == 501
```
Comment (edwardsph) [inline protocol/read-write-resource/method-not-allowed.feature:14]:
```suggestion
    Then match [400, 405, 501] contains response.status
```

**Review by edwardsph — APPROVED** [review]
LGTM. Just wondered about matching style for other places where there are > 2 options.
